ai2thor/controller.py

```diff
@@ -532,6 +558,7 @@ def __exit__(self, *args):
         self.stop()
 
     @property
+    @lru_cache()
```

Comment:
Caching is already happening in the `scenes_in_build` function through the use of the private `_scenes_in_build` attribute so I don't see the benefit of adding a lru cache?

Reply:
I think using `controller._scenes_in_build` is quite dangerous. For one, it is initialized to `None` and so if you call `self._scenes_in_build`, it's possible that the value is none.

Of course, there are workarounds, such as checking `if not self._scenes_in_build: <generate scenes_in_build>` each time one may ever need to call `controller.scenes_in_build`, but this is an uglier workarounds than just using `controller.scenes_in_build`, which will always produce the correct output (of the scenes in the build). We can then cache this value because it will not change.

Reply:
Moreover, `self.scenes_in_build` has been used all over the place, and to the best of my knowledge, `self._scenes_in_build` is basically used nowhere.

Reply:
Actually, nevermind. I see what you're saying here.

Reply:
I think I might not have been clear. In the current `scenes_in_build` function, the first thing it does is check if `self._scenes_in_build is None` and, if it's not, returns `self._scenes_in_build`. Otherwise it computes a value for `self._scenes_in_build` and then returns it. I.e. this function _is already_ doing caching. The `@lru_cache()` call is thus redundant. If you want to remove `self._scenes_in_build` and just use `@lru_cache()` that would also be fine but, in that case, what you really want is `@lazy_property` (see how this is implemented in pytorch) rather than an `lru_cache` as a lru cache has to do a dictionary lookup.

Reply:
Added the initial fix by just removing lru_cache for now. Potentially can slightly optimize it in the future, but probably not worth doing at the moment.